ext/snmp/snmp.c

```diff
@@ -1374,6 +1374,28 @@ PHP_FUNCTION(snmp_set_oid_output_format)
 }
 /* }}} */
 
+/* {{{ Set the string output format. */
+PHP_FUNCTION(snmp_set_string_output_format)
+{
+	zend_long a1;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &a1) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	switch((int) a1) {
```

Comment:
Is it possible to omit this cast? Otherwise you miss the ValueError when the bottom half of the integer matches a constant, but the top half does not.

Reply:
@nikic, I pulled the prototype for this function whole cloth from `snmp_set_oid_value_format()`, are you suggesting it should be changed as well.

Reply:
You're right, I've adjusted that function in https://github.com/php/php-src/commit/b20362c24b673baf4143410e25181d05f4b940d2.

Reply:
Okay, updated per your comments.